lib/app/features/zodiac/domain/usecases/stealth_security_action.dart

```diff
@@ -63,9 +63,20 @@ class StealthSecurityAction {
   }
 
   Future<UserLocationEntity> _getCurrentLocatin() async {
-    return _locationService
-        .currentLocation()
-        .then((v) => v.getOrElse(() => const UserLocationEntity())!);
+    await _locationService.requestPermission(
+      title: 'O guardião precisa da sua localização',
+      description: RequestLocationPermissionContentWidget(),
+    );
+
+    final hasPermission = await hasLocationPermission();
+
+    if (hasPermission) {
+      return _locationService.currentLocation().then((v) {
+        return v.getOrElse(() => const UserLocationEntity())!;
+      });
+    }
+
+    return const UserLocationEntity();
```

Comment:
Pelo que entendi se não tiver a permissão ele manter o comportamento anterior. É isso mesmo?

Não sei se isso foi explicitamente conversado, mas será que se não tivermos a localização deveríamos "para o fluxo" e não enviar o SMS (e avisar que não estamos enviando porque não temos a localização)?

Ou talvez enviar a localização para um "endereço default"?
Acho que vale conversar sobre isso lá no card do trello, caso ainda não tenha sido conversado. :)

Reply:
Opa, só pra confirmar. O front vai continuar mandando o Lat Long zerado se não for aprovado e o back vai mandar a mensagem pro guardião explicando que n foi possível acessar a localização.

Reply:
Pessoal, vou finalizar esse conversation, qualquer coisa a gente abre outro ok?